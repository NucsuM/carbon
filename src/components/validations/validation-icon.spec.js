import React from "react";
import TestRenderer from "react-test-renderer";
import { mount } from "enzyme";
import ValidationIcon from "./validation-icon.component";
import ValidationIconStyle from "./validation-icon.style";
import {
  InputContext,
  InputGroupContext,
} from "../../__internal__/input-behaviour";
import ClassicTheme from "../../style/themes/classic";
import "jest-styled-components";
import Icon from "../icon";

describe("ValidationIcon", () => {
  it.each([
    [{ error: "Message" }, "error"],
    [{ warning: "Message" }, "warning"],
    [{ info: "Message" }, "info"],
  ])(
    "renders with a proper icon if validation prop is passed as string",
    (validation, iconType) => {
      const wrapper = mount(<ValidationIcon {...validation} />);
      expect(wrapper.find(ValidationIconStyle).prop("validationType")).toEqual(
        iconType
      );
    }
  );

  it.each([[{ error: true }], [{ warning: true }], [{ info: true }]])(
    "does not render any icon if validation prop is a string",
    (validation) => {
      const wrapper = mount(<ValidationIcon {...validation} />);
      expect(wrapper.find(ValidationIconStyle).exists()).toBe(false);
    }
  );

  it("renders with an icon with classic styling", () => {
    const wrapper = TestRenderer.create(
      <ValidationIconStyle theme={ClassicTheme} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('"tooltipPosition" and "tooltipAlign" props in its icon should be "right" and "center" respectively', () => {
    const wrapper = mount(<ValidationIcon error="error" />);
    const iconProps = wrapper.find(Icon).props();
    expect(iconProps.tooltipPosition).toBe("right");
    expect(iconProps.tooltipAlign).toBe("center");
  });

  it('does not pass "tooltipPosition" and "tooltipAlign" props to its icon for the classic theme', () => {
    const wrapper = mount(
      <ValidationIcon error="error" theme={ClassicTheme} />
    );
    const iconProps = wrapper.find(Icon).props();
    expect(iconProps.tooltipPosition).toBe(undefined);
    expect(iconProps.tooltipAlign).toBe(undefined);
  });

  it("shows the tooltip if input context has focus", () => {
    const icon = renderWithInputContext({ hasFocus: true }).find("Icon");
    expect(icon.props().tooltipVisible).toEqual(true);
  });

  it("shows the tooltip if input context has mouse over", () => {
    const icon = renderWithInputContext({ hasMouseOver: true }).find("Icon");
    expect(icon.props().tooltipVisible).toEqual(true);
  });

  it("shows the tooltip if input group context has focus", () => {
    const icon = renderWithInputContext({}, { hasFocus: true }).find("Icon");
    expect(icon.props().tooltipVisible).toEqual(true);
  });

  it("shows the tooltip if input group context has mouse over", () => {
    const icon = renderWithInputContext({}, { hasMouseOver: true }).find(
      "Icon"
    );
    expect(icon.props().tooltipVisible).toEqual(true);
  });
});

function renderWithInputContext(
  inputContextValue = {},
  inputGroupContextValue = {}
) {
  return mount(
    <InputContext.Provider value={inputContextValue}>
      <InputGroupContext.Provider value={inputGroupContextValue}>
        <ValidationIcon error="Message" />
      </InputGroupContext.Provider>
    </InputContext.Provider>
  );
}
