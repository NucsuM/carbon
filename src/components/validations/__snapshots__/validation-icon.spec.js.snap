// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationIcon renders with an icon with classic styling 1`] = `
.c0 {
  background: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0;
  margin-bottom: 0;
  border: 0;
  outline: none;
}

.c0 .c1:before {
  color: #C7384F;
}

.c0 .c1:focus {
  outline: none;
}

<div
  className="c0"
/>
`;
